recorder.go

```diff
@@ -0,0 +1,112 @@
+package veneur
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+)
+
+// Recorder is a type-safe interface for emitting metrics.
+type Recorder struct {
+	Statsd *statsd.Client
+}
+
+// NewRecorder makes a new Recorder instance from a Config.
+func NewRecorder(conf Config) (*Recorder, error) {
+	statsd, err := statsd.NewBuffered(conf.StatsAddress, 1024)
+	if err != nil {
+		return nil, err
+	}
+	statsd.Namespace = "veneur."
+	statsd.Tags = append(conf.Tags, "veneurlocalonly")
+
+	return &Recorder{
+		Statsd: statsd,
+	}, nil
+}
+
+// NewProxyRecorder makes a new Recorder instance from a ProxyConfig.
+func NewProxyRecorder(conf ProxyConfig) (*Recorder, error) {
+	statsd, err := statsd.NewBuffered(conf.StatsAddress, 1024)
+	if err != nil {
+		return nil, err
+	}
+	statsd.Namespace = "veneur."
+
+	return &Recorder{
+		Statsd: statsd,
+	}, nil
+}
+
+// DurationSinceStart is a convenience method for getting a duration in
+// nanoseconds given a start time.
+func DurationSinceStart(start time.Time) float64 {
+	return float64(time.Since(start).Nanoseconds())
+}
+
+// LocalMetricsFlushCounts reports the counts of
+// Counters, Gauges, LocalHistograms, LocalSets, and LocalTimers
+// as metrics. These are shared by both global and local flush operations.
+// It does *not* report the totalHistograms, totalSets, or totalTimers
+// because those are only performed by the global veneur instance.
+// It also does not report the total metrics posted, because on the local veneur,
+// that should happen *after* the flush-forward operation.
+func (r *Recorder) LocalMetricsFlushCounts(ms metricsSummary) {
+	r.Statsd.Count("worker.metrics_flushed_total", int64(ms.totalCounters), []string{"metric_type:counter"}, 1.0)
```

Comment:
I'd recommend making each of these strings a `const` at the top of the function, to prevent typos.